### PR TITLE
Enable new style quota configuration in testing

### PR DESCRIPTION
**Pull request by mandarjog**
Mixer2 quota requires a different type of config.
This updates the config and uses a Mixer build that supports that config.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: samples/apps/bookinfo/rules/mixer-rule-ratings-ratelimit.yaml [hunk 1]
```diff
@@ -1,8 +1,37 @@
...
-        expiration: 1s
+apiVersion: "config.istio.io/v1alpha2"
+kind: memquota
+metadata:
+  name: handler
+  namespace: istio-config-default
+spec:
+  quotas:
+  - name: requestcount.quota.istio-config-default
+    max_amount: 5000
```
Comment (kyessenov) [inline samples/apps/bookinfo/rules/mixer-rule-ratings-ratelimit.yaml]: nit: JSON normal form is `maxAmount`, although proto parser is lenient.

File: samples/apps/bookinfo/rules/mixer-rule-ratings-ratelimit.yaml [hunk 2]
```diff
@@ -1,8 +1,37 @@
...
+    valid_duration: 1s
+    overrides:
+    - dimensions:
+        destination: ratings
```
Comment (kyessenov) [inline samples/apps/bookinfo/rules/mixer-rule-ratings-ratelimit.yaml]: Might be worth adding a comment what this means

File: samples/apps/bookinfo/rules/mixer-rule-ratings-ratelimit.yaml [hunk 3]
```diff
@@ -1,8 +1,37 @@
...
+---
+apiVersion: "config.istio.io/v1alpha2"
+kind: quota
+metadata:
+  name: requestcount
+  namespace: istio-config-default
+spec:
+  dimensions:
+    source: source.labels["app"] | "unknown"
+    destination: destination.labels["apps"] | "unknown"
```
Comment (kyessenov) [inline samples/apps/bookinfo/rules/mixer-rule-ratings-ratelimit.yaml]: `destination.name` also works, right?

File: samples/apps/bookinfo/rules/mixer-rule-standard-attributes.yaml [hunk 4]
```diff
@@ -47,8 +47,30 @@ spec:
...
+        valueType: INT64
+      connection.sent.bytes_total:
+        valueType: INT64
+      connection.duration:
+        valueType: DURATION
+      context.protocol:
+        valueType: STRING
+      context.timestamp:
+        valueType: TIMESTAMP
+      context.time:
```
Comment (kyessenov) [inline samples/apps/bookinfo/rules/mixer-rule-standard-attributes.yaml]: Why both time and timestamp? This also looks like a more consequential change than quota update.

File: samples/apps/bookinfo/rules/mixer-rule-standard-attributes.yaml [hunk 5]
```diff
@@ -84,4 +106,16 @@ spec:
         valueType: STRING
       target.serviceAccount:
         valueType: STRING
+      destination.ip:
```
Comment (kyessenov) [inline samples/apps/bookinfo/rules/mixer-rule-standard-attributes.yaml]: Please add `destination.port` if you are at it.

File: samples/apps/bookinfo/rules/mixer-rule-standard-attributes.yaml [hunk 6]
```diff
@@ -47,8 +47,30 @@ spec:
...
+        valueType: INT64
+      connection.sent.bytes:
+        valueType: INT64
+      connection.sent.bytes_total:
+        valueType: INT64
+      connection.duration:
+        valueType: DURATION
+      context.protocol:
+        valueType: STRING
+      context.timestamp:
```
Comment (kyessenov) [inline samples/apps/bookinfo/rules/mixer-rule-standard-attributes.yaml]: the casing is all over place `serviceAccount`, `user-agent` and `bytes_total`?

File: samples/apps/bookinfo/rules/mixer-rule-ratings-ratelimit.yaml (hunk 3 above)
Comment (mandarjog) [reply samples/apps/bookinfo/rules/mixer-rule-ratings-ratelimit.yaml]: `target.name: istio-ingress-2398531633-tj53n`  is the pod hostname so you can't write rules based on it.

Comment (kyessenov) [reply samples/apps/bookinfo/rules/mixer-rule-ratings-ratelimit.yaml]: Interesting, that sounds like a bug. "name" should "istio-ingress" here, the short name of the service.

**Comment by ldemailly** [comment]
what do we need to do so people type something instead of "<PLEASE PUT RELEASE-NOTE HERE, PUT NONE if NO RELEASE-NOTE>"

File: install/kubernetes/addons/grafana.yaml [hunk 7]
```diff
@@ -26,7 +26,7 @@ spec:
     spec:
       containers:
       - name: grafana
-        image: gcr.io/istio-testing/grafana:1bc30a23190aa58635d02ff7fd31bf74de0d011e
+        image: gcr.io/istio-testing/grafana:1af0c4d990ff6bb881c0243b802ed4a8d96f1a5a
```
Comment (ldemailly) [inline install/kubernetes/addons/grafana.yaml]: ^^^

File: samples/apps/bookinfo/rules/mixer-rule-ratings-ratelimit.yaml (hunk 2 above)
Comment (mandarjog) [reply samples/apps/bookinfo/rules/mixer-rule-ratings-ratelimit.yaml]: done

File: samples/apps/bookinfo/rules/mixer-rule-standard-attributes.yaml (hunk 6 above)
Comment (mandarjog) [reply samples/apps/bookinfo/rules/mixer-rule-standard-attributes.yaml]: Those I have just copied over from the API repo, let's see if we can fix it there.

File: samples/apps/bookinfo/rules/mixer-rule-standard-attributes.yaml (hunk 4 above)
Comment (mandarjog) [reply samples/apps/bookinfo/rules/mixer-rule-standard-attributes.yaml]: copied from the API repo.

File: samples/apps/bookinfo/rules/mixer-rule-standard-attributes.yaml (hunk 5 above)
Comment (mandarjog) [reply samples/apps/bookinfo/rules/mixer-rule-standard-attributes.yaml]: needs to be done in the API repo first.

File: install/kubernetes/addons/grafana.yaml (hunk 7 above)
Comment (mandarjog) [reply install/kubernetes/addons/grafana.yaml]: This PR needs at least the given level of Mixer. Once that [PR](https://github.com/istio/mixer/pull/1256) is merged, I will remove the specific sha. 
I will not merge this PR with this sha.

File: samples/apps/bookinfo/rules/mixer-rule-standard-attributes.yaml (hunk 6 above)
Comment (kyessenov) [reply samples/apps/bookinfo/rules/mixer-rule-standard-attributes.yaml]: Can you point me to the source of truth? I'll file an issue.

Comment (douglas-reid) [reply samples/apps/bookinfo/rules/mixer-rule-standard-attributes.yaml]: we need a pass at this anyway in istio/api.

**Comment by mandarjog** [comment]
```
Tests Summary
PASSED: //tests/e2e/tests/mixer:go_default_test 
PASSED: //tests/e2e/tests/bookinfo:go_default_test 
```
This PR with the correct Mixer build passes.

**Comment by mandarjog** [comment]
@rshriram @ldemailly let's merge this please.
This is required for tests to pass at head of mixer.
It includes commits from #775

**Comment by ldemailly** [comment]
failed with #777

File: tests/e2e/tests/mixer/mixer_test.go [hunk 8]
```diff
@@ -461,21 +460,21 @@ func TestRateLimit(t *testing.T) {
...
-		t.Fatalf("Could not get metrics from prometheus: %v", err)
+		fatalf(t, "Could not get metrics from prometheus: %v", err)
 	}
 	glog.Infof("promvalue := %s", value.String())
 
 	got, err := vectorValue(value, map[string]string{responseCodeLabel: "429"})
 	if err != nil {
 		t.Logf("prometheus values for request_count:\n%s", promDump(promAPI, "request_count"))
-		t.Fatalf("Could not find rate limit value: %v", err)
+		fatalf(t, "Could not find rate limit value: %v", err)
```
Comment (ldemailly) [inline tests/e2e/tests/mixer/mixer_test.go]: this should not be fatal, should be errorf + got =0

Comment (mandarjog) [reply tests/e2e/tests/mixer/mixer_test.go]: let's not change it in this PR.

**Comment by douglas-reid** [comment]
private cluster testing:
```
****************************************************
Tests Summary
PASSED: //tests/e2e/tests/mixer:go_default_test
PASSED: //tests/e2e/tests/bookinfo:go_default_test
```

**Pull request merged**